docs/proposals/2170-kubeflow-training-v2/README.md

```diff
@@ -337,17 +337,16 @@ type TrainJobStatus struct {
 	// Conditions for the TrainJob.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// ReplicatedJobsStatus tracks the number of Jobs for each replicatedJob in TrainJob.
-	ReplicatedJobsStatus []jobsetv1alpha2.ReplicatedJobStatus `json:"replicatedJobsStatus,omitempty"`
+	// JobsStatus tracks the child Jobs in TrainJob.
+	JobsStatus []JobStatus `json:"jobsStatus,omitempty"`
 }
 
-type ReplicatedJobStatus struct {
-	// Name of the ReplicatedJob.
+type JobStatus struct {
```

Comment:
Might we have a use case where the training job is not represented by the Job but by some other entity (e.g. NodeSet in Slurm) ?

Reply:
I guess that we need to define what we want to call a set of actual deployment resources.
Currently, once we submit TrainJob and Runtimes (TraininigRuntime / ClusterTrainingRuntime), the operator creates a set of resources like JobSet, PodGroup, HPA and so on...

Can we call a set of resources Kubeflow TraininigCluster, TrainingSet, TrainingService, or TrainingStack something? Do you have any idea?

For example, if we choice the TraininigCluster, we can replace this `JobStatus` with `TrainingClusterStatus`.
WDYT? @andreyvelich

Reply:
Given that you want to expose Job status as part of runtimes, I think it's fine to keep the JobStatus field in TrainJob: https://github.com/kubeflow/training-operator/pull/2296#discussion_r1809381447.
In the future, if we want to introduce the stateful `TrainingCluster`, maybe we should create another CRD.

Reply:
> Given that you want to expose Job status as part of runtimes, I think it's fine to keep the JobStatus field in TrainJob: https://github.com/kubeflow/training-operator/pull/2296#discussion_r1809381447.

That sounds good to me. 
For the runtime interface and Kubeflow Job Pipeline Framework, I will add a detailed design to this KEP in a separate PR.